config: add vocabulary datastreams

ppanero opened the pull request:
- requires https://github.com/inveniosoftware/invenio-rdm-records/pull/832
- closes #1169 

**Config**
```shell
$ invenio shell

In [2]: current_app.config["VOCABULARIES_DATASTREAM_READERS"]
Out[2]:
{'orcid-http': invenio_vocabularies.contrib.names.datastreams.OrcidHTTPReader,
 'tar': invenio_vocabularies.datastreams.readers.TarReader,
 'yaml': invenio_vocabularies.datastreams.readers.YamlReader}

In [3]: current_app.config["VOCABULARIES_DATASTREAM_TRANSFORMERS"]
Out[3]: {'xml': invenio_vocabularies.datastreams.transformers.XMLTransformer, 
'orcid-xml': invenio_vocabularies.contrib.names.datastreams.OrcidXMLTransformer}

In [4]: current_app.config["VOCABULARIES_DATASTREAM_WRITERS"]
Out[4]:
{'names-service': invenio_vocabularies.contrib.names.datastreams.NamesServiceWriter,
 'service': invenio_vocabularies.datastreams.writers.ServiceWriter,
 'yaml': invenio_vocabularies.datastreams.writers.YamlWriter}
 ```

**Import names**
```shell
$ invenio vocabularies import -v names -f ../../../zenodo/zenodo-rdm/app_data/vocabularies-future.yaml
Vocabulary names imported. Total items 132.
132 items succeeded, 0 contained errors.
```

**Sample search**

```shell
$ curl -k -XGET https://127.0.0.1:5000/api/names\?size\=2 | jq .
{
  "hits": {
    "hits": [
      {
        "revision_id": 1,
        "identifiers": [
          {
            "scheme": "orcid",
            "identifier": "0000-0002-4406-428X"
          }
        ],
        "links": {
          "self": "https://127.0.0.1:5000/api/names/a3yxy-p5v97"
        },
        "id": "a3yxy-p5v97",
        "affiliations": [],
        "created": "2022-01-20T08:23:17.855530+00:00",
        "family_name": "ARTINI",
        "updated": "2022-01-20T08:23:17.867574+00:00",
        "given_name": "MICHELE",
        "name": "ARTINI, MICHELE"
      },
      {
        "revision_id": 1,
        "identifiers": [
          {
            "scheme": "orcid",
            "identifier": "0000-0003-2244-4916"
          }
        ],
        "links": {
          "self": "https://127.0.0.1:5000/api/names/q1cj3-7ph87"
        },
        "id": "q1cj3-7ph87",
        "affiliations": [],
        "created": "2022-01-20T08:23:20.682685+00:00",
        "family_name": "Alexiou",
        "updated": "2022-01-20T08:23:20.694635+00:00",
        "given_name": "Giorgos",
        "name": "Alexiou, Giorgos"
      }
    ],
    "total": 132
  },
  "sortBy": "name",
  "links": {
    "self": "https://127.0.0.1:5000/api/names?page=1&size=2&sort=name",
    "next": "https://127.0.0.1:5000/api/names?page=2&size=2&sort=name"
  }
}
```

ppanero commented on invenio_app_rdm/config.py:
```diff
@@ -494,6 +498,28 @@ def files_rest_permission_factory(obj, action):
+# Invenio-Vocabularies
+# =============
+
+VOCABULARIES_DATASTREAM_READERS = {
```
@slint should I do this, or something like:

```python
if VOCABULARIES_DATASTREAM_READERS:
    VOCABULARIES_DATASTREAM_READERS["orcid-http"] = OrcidHTTPReader
else:
    VOCABULARIES_DATASTREAM_READERS= {
        "orcid-http": OrcidHTTPReader,
    }
```

in order to avoid missing "default" ones from `invenio-vocabularies... There was a similar method I recall for `invenio-oauthclient` config, but I did not manage to find it...

slint replied:
Sometimes I think we just make a "deepcopy" and update the `dict`... that way we don't miss any default config that comes out of the box. So something like:

```python
from invenio_vocabularies.config import VOCABULARIES_DATASTREAM_READERS

VOCABULARIES_DATASTREAM_READERS = {
    **VOCABULARIES_DATASTREAM_READERS,
    "orcid-http": OrcidHTTPReader,
    ...
}
```

ppanero replied:
Done :)

ppanero commented:
@slint
- fixed the config according to what's discussed in the comment above
- bump inveno-rdm-records version, to include the names integration release
- added a bit more info in the PR description

slint commented on invenio_app_rdm/config.py:
```diff
@@ -494,6 +498,27 @@ def files_rest_permission_factory(obj, action):
+
+VOCABULARIES_DATASTREAM_READERS = {
+    **VOCABULARIES_DATASTREAM_READERS,
+    "orcid-http": OrcidHTTPReader,
```
Minor thought here (to be shelved when we revisit contrib modules config injection), would it make sense to expose in each invenio-vocabularies contrib a config variable for `readers/transformers/writers`, so that one can e.g. easily get all of the latest config like so:

```python
from invenio_vocabularies.config import VOCABULARIES_DATASTREAM_TRANSFORMERS
import invenio_vocabularies.contrib.names.config as names_config

VOCABULARIES_DATASTREAM_TRANSFORMERS = {
    **VOCABULARIES_DATASTREAM_TRANSFORMERS,
    **names_config.VOCABULARIES_DATASTREAM_TRANSFORMERS,
}
```

ppanero replied:
This actually makes a lot of sense, and I think is already possible... I'll do the changes
